src/Bicep.Core/Semantics/SemanticModel.cs

```diff
@@ -562,7 +562,7 @@ private IEnumerable<IDiagnostic> GetAdditionalParamsSemanticDiagnostics()
             return
                 // get diagnostics relating to missing parameter assignments or declarations
                 GatherParameterMismatchDiagnostics(semanticModel)
-                    .Concat(GatherMissingRequiredExtensionConfigAssignmentDiagnostics(semanticModel))
+                    .Concat(semanticModel is SemanticModel semanticModelImpl ? GatherMissingRequiredExtensionConfigAssignmentDiagnostics(semanticModelImpl) : [])
```

Comment:
nit: It might be cleaner to put the logic in `GatherMissingRequiredExresionConfigAssignmentDiagnostcis`.